Sources/PCRoomSubscription.swift

```diff
@@ -2,67 +2,26 @@ import Foundation
 import PusherPlatform
 
 public final class PCRoomSubscription {
-    public var delegate: PCRoomDelegate?
-    let resumableSubscription: PPResumableSubscription
-    public var logger: PPLogger
-    let basicMessageEnricher: PCBasicMessageEnricher
+    let messageSubscription: PCMessageSubscription
+    let cursorSubscription: PCCursorSubscription
+
+    // TODO: Maybe the RoomSubscription should just store the delegate?
+    public var delegate: PCRoomDelegate? {
+        get {
+            return messageSubscription.delegate
+        }
+    }
```

Comment:
Wouldn't it be better to provide the delegate in the constructor instead of having a `var`?

Reply:
Potentially - I hadn't made my mind up yet, hence the TODO: https://github.com/pusher/chatkit-swift/pull/69/files#diff-6ae7c192bfbc593d33d9f6769b3f8ea3R8 

I was thinking of making the delegate a non-optional property of the two subscriptions as well, but hadn't fully thought through whether or not that was too big of a breaking change. 

I'll take a proper look and think about it 👍